store/sqlstore/channel_store.go

```diff
@@ -3100,3 +3377,242 @@ func (s SqlChannelStore) GroupSyncedChannelCount() (int64, *model.AppError) {
 
 	return count, nil
 }
+
+type sidebarCategoryForJoin struct {
+	model.SidebarCategory
+	ChannelId string
+}
+
+func (s SqlChannelStore) CreateSidebarCategory(userId, teamId string, newCategory *model.SidebarCategoryWithChannels) (*model.SidebarCategoryWithChannels, *model.AppError) {
+	transaction, err := s.GetMaster().Begin()
+	if err != nil {
+		return nil, model.NewAppError("SqlChannelStore.CreateSidebarCategory", "store.sql_channel.sidebar_categories.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	defer finalizeTransaction(transaction)
+
+	categoryId := model.NewId()
+	category := &model.SidebarCategory{
+		DisplayName: newCategory.DisplayName,
+		Id:          categoryId,
+		UserId:      userId,
+		TeamId:      teamId,
+		SortOrder:   0,
+		Type:        model.SidebarCategoryCustom,
+	}
+	if err = transaction.Insert(category); err != nil {
+		return nil, model.NewAppError("SqlPostStore.CreateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	var channels []interface{}
+	runningOrder := 0
+	for _, channelID := range newCategory.Channels {
+		channels = append(channels, &model.SidebarChannel{
+			ChannelId:  channelID,
+			CategoryId: categoryId,
+			SortOrder:  int64(runningOrder),
+			UserId:     userId,
+		})
+		runningOrder += 10
+	}
+	if err = transaction.Insert(channels...); err != nil {
+		return nil, model.NewAppError("SqlPostStore.CreateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	if err = transaction.Commit(); err != nil {
+		return nil, model.NewAppError("SqlChannelStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.commit_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	return &model.SidebarCategoryWithChannels{
+		SidebarCategory: *category,
+		Channels:        newCategory.Channels,
+	}, nil
+}
+
+func (s SqlChannelStore) GetSidebarCategory(userId, teamId, categoryId string) (*model.SidebarCategoryWithChannels, *model.AppError) {
+	var categories []*sidebarCategoryForJoin
+	if _, err := s.GetReplica().Select(&categories, "SELECT *, SidebarChannels.ChannelId FROM SidebarCategories, SidebarChannels WHERE SidebarCategories.UserId=:userId AND SidebarCategories.TeamId=:teamId AND SidebarChannels.CategoryId=SidebarCategories.Id AND SidebarCategories.Id=:categoryId ORDER BY SidebarChannels.SortOrder ASC", map[string]interface{}{"userId": userId, "teamId": teamId, "categoryId": categoryId}); err != nil {
```

Comment:
Would it be possible to use squirrel for this and the next SQL query ? Might be cleaner if at least the read queries use squirrel.